core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,109 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+public class LocalFeedUpdater {
+
+    private static long getFileDuration(File f) {
+        MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+        mediaMetadataRetriever.setDataSource(f.getAbsolutePath());
+        String durationStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+        return Long.parseLong(durationStr);
+    }
+
+    /** Starts the import process. */
+    public static void startImport(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        if (!f.isDirectory()) {
+            throw new RuntimeException("invalid path");
+        } else {
+            startImportDirectory(uri, context);
+        }
+    }
+
+    private static void startImportDirectory(Uri uri, Context context) {
+        //create a feed object for this directory
+        File f = new File(uri.getPath());
+        String dirUrl = uri.toString();
+        Feed dirFeed = new Feed(dirUrl, null, "Local directory (" + dirUrl + ")");
+        dirFeed.setItems(new ArrayList<>()); //this seems useless
+        //find the feed for this directory (if it exists), or create one
+        dirFeed = DBTasks.updateFeed(context, dirFeed)[0];
+
+        //find relevant files and create items for them
+        File[] itemFiles = f.listFiles(
+                new FilenameFilter() {
+                    @Override
+                    public boolean accept(File file, String s) {
+                        String m = getMimeType(s);
+                        return m != null && (m.startsWith("audio/") || m.startsWith("video/"));
+                    }
+                });
+        List<FeedItem> newItems = new ArrayList<>();
```

Comment:
I think you can use dirFeed.getItems() here to automatically include existing items. With that, you don't need to do anything in the loop below if the item is already there

Reply:
You mean append to the list returned by `getItems()`? I guess I didn't do that because I wasn't sure I was getting a "reference" rather than a copy...

Reply:
> You mean append to the list returned by getItems()?

Yes. I am not totally sure but I think it should work.